kong/plugins/aws-lambda/iam-ec2-credentials.lua

```diff
@@ -9,14 +9,39 @@ local kong = kong
 local METADATA_SERVICE_PORT = 80
 local METADATA_SERVICE_REQUEST_TIMEOUT = 5000
 local METADATA_SERVICE_HOST = "169.254.169.254"
-local METADATA_SERVICE_URI = "http://" .. METADATA_SERVICE_HOST .. ":" .. METADATA_SERVICE_PORT ..
+local METADATA_SERVICE_TOKEN_URI = "http://" .. METADATA_SERVICE_HOST .. ":" .. METADATA_SERVICE_PORT ..
+                             "/latest/api/token"
+local METADATA_SERVICE_IAM_URI = "http://" .. METADATA_SERVICE_HOST .. ":" .. METADATA_SERVICE_PORT ..
                              "/latest/meta-data/iam/security-credentials/"
 
 
 local function fetch_ec2_credentials()
   local client = http.new()
   client:set_timeout(METADATA_SERVICE_REQUEST_TIMEOUT)
-  local role_name_request_res, err = client:request_uri(METADATA_SERVICE_URI)
+
+  local token_request_res, err = client:request_uri(METADATA_SERVICE_TOKEN_URI, {
+    method = "PUT",
+    headers = {
+      ["X-aws-ec2-metadata-token-ttl-seconds"] = "60",
+    },
+  })
+
+  if not token_request_res then
+    return nil, "Could not fetch IMDSv2 token from metadata service: " .. tostring(err)
```

Comment:
if this request returns a failure, of any kind, should it then abort?

Essentially it means that IDMSv2 is required, if coded like that. I think it would be safer/defensive to continue without a token, using the older IDMS service